webserver/WebContent/app/controllers/owner-controller.js

```diff
@@ -0,0 +1,285 @@
+                size: size,
+                resolve: {
+                    items: function () {
+                        return ownerVM.assignReservation;
+                    }
+                }
+            });
+
+            modalInstance.result.then(function (data) {
+                //owner might have changed or assigned tables. So fetch reservations again.
+                ownerVM.fetchAllReservations();
+            }, function () {
+                //Modal dismissed
+            });
+        };
+
+        //fetch all future reservations.
+        ownerVM.fetchAllReservations = function () {
+
+            ownerService.fetchAllReservations().then(function (data) {
+                ownerVM.reservations = data;
+                ownerVM.page.totalItems = ownerVM.reservations.length;
+                //       console.dir(data);
+            }, function (err) {
+                console.dir(err)
+            })
+        };
+        //get the restaurant profile. contact timings etc
+        ownerVM.getProfile = function () {
+
+            ownerService.getProfile().then(function (data) {
+
+                ownerVM.profile = data;
+
+
+                //   console.dir(data);
+            }, function (err) {
+                console.dir(err)
+            })
+        };
+        //get the current seating area details
+        ownerVM.getSeatingArea = function () {
+
+            ownerService.getSeatingArea().then(function (data) {
+
+                ownerVM.seatingArea = data;
+
+                // console.dir(data);
+            }, function (err) {
+                console.dir(err)
+            })
+        };
+        //get info of every customer that has made a reservation
+        ownerVM.getAllCustomers = function () {
+
+            ownerService.getAllCustomers().then(function (data) {
+                ownerVM.customers = data;
+                //   console.dir(data);
+            }, function (err) {
+                console.dir(err)
+            })
+        };
+        //save the restaurant profile
+        ownerVM.saveProfile = function () {
+
+
+            ownerService.saveProfile(ownerVM.profile).then(function (data) {
+
+                ownerVM.modalMessage = "Profile was saved";
+                ownerVM.openMessageModal();
+
+            }, function (err) {
+                console.dir(err)
+            })
+        };
+        //format the data so that owner can view it
+        ownerVM.formatReservation = function () {
+
+            ownerVM.showReservation = true;
+            ownerVM.editReservation = true;
+            var t = new Date(ownerVM.reservation.time);
+            ownerVM.reservation.date = new Date();
+            ownerVM.reservation.date.setMilliseconds(t.getMilliseconds());
+
+            ownerVM.reservation.rtime = new Date(1970, 0, 1, t.getHours(), t.getMinutes(), 0);
+        };
+        //create a new reservation
+        ownerVM.createReservation = function () {
```

Comment:
repetitive functionality should be extracted out into a service or a factory and used via that.